pkg/clients/s3/bucketclients/clients.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bucketclients
+
+import (
+	"context"
+
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+
+	"github.com/crossplane/provider-aws/apis/s3/v1beta1"
+	"github.com/crossplane/provider-aws/pkg/clients/s3"
+)
+
+// BucketResource is the interface all bucket sub-resources must conform to
+type BucketResource interface {
+	Observe(ctx context.Context, bucket *v1beta1.Bucket) (ResourceStatus, error)
+	Create(ctx context.Context, bucket *v1beta1.Bucket) (managed.ExternalUpdate, error)
```

Comment:
It seems like `Create` is called for both `Create` and `Update` purposes. Would it make sense to rename it to `Put` or `CreateOrUpdate` or `Upsert` ?